src/python/pants/backend/nfpm/target_types.py

```diff
@@ -142,6 +151,47 @@ class NfpmApkPackage(NfpmPackageTarget):
     )
 
 
+ARCHLINUX_FIELDS = (
+    NfpmPackageNameField,
+    NfpmArchField,
+    # version fields (archlinux does NOT get: version_metadata)
+    NfpmVersionField,
+    NfpmVersionSchemaField,
+    NfpmVersionPrereleaseField,
+    NfpmVersionReleaseField,
+    NfpmVersionEpochField,
+    # other package metadata fields
+    NfpmHomepageField,
+    NfpmLicenseField,
+    NfpmArchlinuxPackagerField,
+    NfpmArchlinuxPkgbaseField,
+    # package relationships
+    NfpmArchlinuxReplacesField,
+    NfpmArchlinuxProvidesField,
+    NfpmArchlinuxDependsField,
+    NfpmArchlinuxConflictsField,
+)
+
+
+class NfpmArchlinuxPackage(NfpmPackageTarget):
+    alias = "nfpm_archlinux_package"
+    core_fields = (
+        *COMMON_NFPM_PACKAGE_FIELDS,
+        *ARCHLINUX_FIELDS,
+        NfpmArchlinuxScriptsField,
```

Comment:
I noticed that the scripts fields are not part of the `PACKAGER_FIELDS` constants. For my own education, why are they not included there?

Reply:
The packager field constants are used to populate the field sets `required_fields`. Then, the field set's `nfpm_config` method loops over `required_fields` to add each field's config. But, the logic for scripts fields is not the same as the logic for the other fields. So, excluding the scripts fields from `required_fields` lets me used different logic for them.

Admittedly, this is an unorthodox use of field sets, but it was the simplest way I came up with for dealing with the vast number of fields.

Reply:
Maybe worth an explanatory comment somewhere? (And one-line comment before the line with the scripts field in these target definitions referring to the explanatory comment.)